src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/Api/UnitTestingRemoteHostClientWrapper.cs

```diff
@@ -3,31 +3,48 @@
 // See the LICENSE file in the project root for more information.
 
 #nullable enable
-#pragma warning disable CS0618 // Type or member is obsolete
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Remote;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
 {
     internal readonly struct UnitTestingRemoteHostClientWrapper
     {
-        internal UnitTestingRemoteHostClientWrapper(RemoteHostClient underlyingObject)
-            => UnderlyingObject = underlyingObject ?? throw new ArgumentNullException(nameof(underlyingObject));
+        internal UnitTestingRemoteHostClientWrapper(RemoteHostClient? underlyingObject)
+            => UnderlyingObject = underlyingObject;
 
-        internal RemoteHostClient UnderlyingObject { get; }
+        internal RemoteHostClient? UnderlyingObject { get; }
 
+        public bool IsDefault => UnderlyingObject == null;
+
+        public static async Task<UnitTestingRemoteHostClientWrapper?> TryGetClientAsync(HostWorkspaceServices services, CancellationToken cancellationToken = default)
```

Comment:
nit: @Cosifne Not specific to this file but some members on these wrapper types are public while others are internal. Not sure why they can't all be internal or all public consistently...

Reply:
Only the underlying object is marked as internal and I think it is because we shouldn't access the Underlying object directly but some other code in Roslyn might need to access that. (but sure, in this case, I feel it could be private)

Reply:
The types need to be internal. THe methods can be public/intenral